src/opsinputs/opsinputs_varobswriter_mod.F90

```diff
@@ -1213,54 +1208,122 @@ end subroutine opsinputs_varobswriter_fillreportflags
-integer(integer64)             :: ChannelCounts(Ob % Header % NumObsLocal)
-integer                        :: iChannel
-integer                        :: iObs
-logical                        :: localUseActualChans
+integer(integer64)              :: NumChannels
+integer(integer64), allocatable :: ChannelIndicesVar(:,:)
+integer(integer64)              :: ChannelIndices(Ob % Header % NumObsLocal, size(channels))
+integer(integer64)              :: ChannelCounts(Ob % Header % NumObsLocal)
+integer                         :: iChannel
+integer                         :: iObs
+real(kind=c_double)             :: MissingDouble
+integer(integer64)              :: array_loop
+integer                         :: offsetsize
+logical                         :: compressChannels
+integer(integer64)              :: varObsSize_loc
+logical                         :: increaseChanArray_loc
 
 ! Body:
-NumChannels = size(Channels)
+
+compressChannels = .true.
+if (present(compressVarChannels)) then
+  compressChannels = compressVarChannels
+end if
+
+varObsSize_loc = 0
+if (present(varObsSize)) then
+  varObsSize_loc = varObsSize
+end if
+
+increaseChanArray_loc = .false.
+if (present(increaseChanArray)) then
+  increaseChanArray_loc = increaseChanArray
+end if
+
+
+MissingDouble = missing_value(0.0_c_double)
+
+NumChannels = size(channels)
 if (NumChannels == 0) return
 
-! Setup for useActualChans for array indices
-localUseActualchans = .false.
-if (Present(UseActualChans)) then
-  localUseActualChans = UseActualChans
+if (increaseChanArray_loc) then
+  NumChannels = varObsSize_loc
+  allocate(ChannelIndicesVar(Ob % Header % NumObsLocal, varObsSize_loc))
+else
+  allocate(ChannelIndicesVar(Ob % Header % NumObsLocal, size(varChannels)))
 end if
 
 call opsinputs_varobswriter_findchannelspassingqc( &
-  Ob % Header % NumObsLocal, ObsSpace, Channels, Flags, ChannelIndices, ChannelCounts)
+  Ob % Header % NumObsLocal, ObsSpace, channels, Flags, ChannelIndices, ChannelCounts)
+
+ChannelIndicesVar(:,:) = 0
+
 if (FillChanNum) then
   call Ops_Alloc(Ob % Header % ChanNum, "ChanNum", Ob % Header % NumObsLocal, Ob % ChanNum, &
                  num_levels = NumChannels)
-
-  if (localUseActualChans) then
-    do iChannel=1, size(Channels)
-      ChannelIndices(:,iChannel) = Channels(iChannel)
-    end do
-    Ob % ChanNum = ChannelIndices
+  if ((varObsSize_loc /= 0)) then
+    if (varObsSize_loc > size(channels)) then
+      if (size(varChannels) > 0 .and. (size(channels) == size(varChannels))) then
+        if (compressChannels) then
+          offsetsize = abs(varChannels(1) - channels(1))
+          Ob % ChanNum = ChannelIndices
+          where (Ob % ChanNum > 0)
+            Ob % ChanNum = Ob % ChanNum + int(offsetsize, kind=integer64)
+          end where
+        else
+          do iChannel=1,  NumChannels
+            if (iChannel > size(varChannels)) then
+              ChannelIndicesVar(:, iChannel) = -32768
```

Comment:
isn't there a missing ops int value somewhere as a parameter?

Reply:
RMDI looks like the one to use.

Reply:
I think IMDI? (Ops_Constants/GenMod_MiscUMScienceConstants.f90)